homeassistant/components/shelly/switch.py

```diff
@@ -276,13 +244,21 @@ def _update_callback(self) -> None:
         super()._update_callback()
 
 
-class RpcRelaySwitch(ShellyRpcEntity, SwitchEntity):
+class RpcRelaySwitch(ShellyRpcAttributeEntity, SwitchEntity):
     """Entity that controls a relay on RPC based Shelly devices."""
 
-    def __init__(self, coordinator: ShellyRpcCoordinator, id_: int) -> None:
-        """Initialize relay switch."""
-        super().__init__(coordinator, f"switch:{id_}")
-        self._id = id_
+    entity_description: RpcSwitchDescription
+
+    def __init__(
+        self,
+        coordinator: ShellyRpcCoordinator,
+        key: str,
+        attribute: str,
+        description: RpcEntityDescription,
+    ) -> None:
+        """Initialize sensor."""
+        super().__init__(coordinator, key, attribute, description)
+        self._attr_unique_id = f"{super().unique_id}"
```

Comment:
This block looks unneeded since it doesn't change the unique id?

Reply:
Indeed, a leftover of the initial approach :blush: